util/deephash/deephash.go

```diff
@@ -136,7 +137,21 @@ func print(w *bufio.Writer, v reflect.Value, visited map[uintptr]bool, scratch [
 		return acyclic
 	case reflect.Slice, reflect.Array:
 		if v.Type().Elem().Kind() == reflect.Uint8 && v.CanInterface() {
-			fmt.Fprintf(w, "%q", v.Interface())
+			if v.CanAddr() {
+				Len := v.Len()
+				w.Write(strconv.AppendInt(scratch[:0], int64(Len), 10))
+				w.WriteByte(':')
+				if Len > 0 {
+					var s []byte
+					sh := (*reflect.SliceHeader)(unsafe.Pointer(&s))
+					sh.Data = v.Addr().Pointer()
+					sh.Len = Len
+					sh.Cap = Len
+					w.Write(s) // unsafe slice escapes, but w is trusted *bufio.Writer
```

Comment:
This is all to handle arrays? I assume for the slice case you could call `v.Bytes()`?

Could we make an argument upstream that `v.Bytes()` should work on uint8 arrays too?

Reply:
I forgot about Bytes. And it would almost work on Arrays too (using reflect.Value.Slice), but Value.Slice seems to allocate, killing the whole plan.